src/python/pants/backend/visibility/glob.py

```diff
@@ -206,7 +206,7 @@ def create(  # type: ignore[misc]
         tags: tuple[str, ...] | None,
     ) -> TargetGlob:
         return cls(
-            type_=Glob.create(type_) if type_ else None,
+            type_=Glob.create(str(type_)) if type_ else None,
```

Comment:
I'm confused. The type sig says `type_` is a `str | None`...

Reply:
it's a lie, it is `str | Registrar | None`. Perhaps should go back and add that in everywhere...

Reply:
Yeah, let's be honest about the type. Then I'll feel better stamping it.

Reply:
didn't like to import the Registrar type, and realize it was easier to just move the type cast to the scope where the data is type hinted as `Any`, applied some more type casts for the other args as well there then, just to be safe.